### ISSUE TITLE
cfssl installation failed in OS X High Sierra

### ISSUE THREAD

**Issue by ngpestelos** [kelseyhightower/kubernetes-the-hard-way#229]
I was going through the instructions on how to setup client tools (`02-client-tools.md`). Line 15 has instructions on how to install `cfssl`, but this unfortunately resulted in a runtime error (see stack trace below). My workaround was to Install `cfssl` using `brew`:

```
Version: 1.2.0
Revision: dev
Runtime: go1.9
```

Stack trace:

```
failed MSpanList_Insert 0xa0f000 0xc09d5d65ac82 0x0 0x0
fatal error: MSpanList_Insert

runtime stack:
runtime.throw(0x6bbbe0, 0x10)
	/usr/local/go/src/runtime/panic.go:530 +0x90 fp=0x7ffeefbff1b0 sp=0x7ffeefbff198
runtime.(*mSpanList).insert(0x9436e8, 0xa0f000)
	/usr/local/go/src/runtime/mheap.go:933 +0x293 fp=0x7ffeefbff1e0 sp=0x7ffeefbff1b0
runtime.(*mheap).freeSpanLocked(0x942ee0, 0xa0f000, 0x100, 0x0)
	/usr/local/go/src/runtime/mheap.go:809 +0x4be fp=0x7ffeefbff248 sp=0x7ffeefbff1e0
runtime.(*mheap).grow(0x942ee0, 0x8, 0x0)
	/usr/local/go/src/runtime/mheap.go:675 +0x2a0 fp=0x7ffeefbff2a0 sp=0x7ffeefbff248
runtime.(*mheap).allocSpanLocked(0x942ee0, 0x1, 0x0)
	/usr/local/go/src/runtime/mheap.go:553 +0x4e3 fp=0x7ffeefbff2f8 sp=0x7ffeefbff2a0
runtime.(*mheap).alloc_m(0x942ee0, 0x1, 0x15, 0x0)
	/usr/local/go/src/runtime/mheap.go:437 +0x119 fp=0x7ffeefbff328 sp=0x7ffeefbff2f8
runtime.(*mheap).alloc.func1()
	/usr/local/go/src/runtime/mheap.go:502 +0x41 fp=0x7ffeefbff358 sp=0x7ffeefbff328
runtime.systemstack(0x7ffeefbff378)
	/usr/local/go/src/runtime/asm_amd64.s:307 +0xab fp=0x7ffeefbff360 sp=0x7ffeefbff358
runtime.(*mheap).alloc(0x942ee0, 0x1, 0x10000000015, 0xee2f)
	/usr/local/go/src/runtime/mheap.go:503 +0x63 fp=0x7ffeefbff3a8 sp=0x7ffeefbff360
runtime.(*mcentral).grow(0x944ae0, 0x0)
	/usr/local/go/src/runtime/mcentral.go:209 +0x93 fp=0x7ffeefbff410 sp=0x7ffeefbff3a8
runtime.(*mcentral).cacheSpan(0x944ae0, 0x93d4e8)
	/usr/local/go/src/runtime/mcentral.go:89 +0x47d fp=0x7ffeefbff450 sp=0x7ffeefbff410
runtime.(*mcache).refill(0xa0b000, 0x15, 0x7ffeefbff4b8)
	/usr/local/go/src/runtime/mcache.go:119 +0xcc fp=0x7ffeefbff488 sp=0x7ffeefbff450
runtime.mallocgc.func2()
	/usr/local/go/src/runtime/malloc.go:642 +0x2b fp=0x7ffeefbff4a8 sp=0x7ffeefbff488
runtime.systemstack(0x7ffeefbff548)
	/usr/local/go/src/runtime/asm_amd64.s:307 +0xab fp=0x7ffeefbff4b0 sp=0x7ffeefbff4a8
runtime.mallocgc(0x180, 0x6558a0, 0x0, 0x800000000)
	/usr/local/go/src/runtime/malloc.go:643 +0x869 fp=0x7ffeefbff588 sp=0x7ffeefbff4b0
runtime.newobject(0x6558a0, 0x93da10)
	/usr/local/go/src/runtime/malloc.go:781 +0x42 fp=0x7ffeefbff5b0 sp=0x7ffeefbff588
runtime.malg(0x8000, 0x93df00)
	/usr/local/go/src/runtime/proc.go:2634 +0x27 fp=0x7ffeefbff5e8 sp=0x7ffeefbff5b0
runtime.mpreinit(0x93e760)
	/usr/local/go/src/runtime/os1_darwin.go:140 +0x1f fp=0x7ffeefbff600 sp=0x7ffeefbff5e8
runtime.mcommoninit(0x93e760)
	/usr/local/go/src/runtime/proc.go:494 +0x105 fp=0x7ffeefbff648 sp=0x7ffeefbff600
runtime.schedinit()
	/usr/local/go/src/runtime/proc.go:434 +0x79 fp=0x7ffeefbff690 sp=0x7ffeefbff648
runtime.rt0_go(0x7ffeefbff6c0, 0x1, 0x7ffeefbff6c0, 0x0, 0x1, 0x7ffeefbff878, 0x0, 0x7ffeefbff880, 0x7ffeefbff8a8, 0x7ffeefbff8bf, ...)
	/usr/local/go/src/runtime/asm_amd64.s:138 +0x132 fp=0x7ffeefbff698 sp=0x7ffeefbff690
```

**Comment by tokiwinter** [comment]
You can just do everything with `openssl` instead. That's what I've done.

**Comment by jodok** [comment]
installing via homebrew `brew install cfssl` works on high sierra as well.

**Comment by anthcor** [comment]
On osx 10.13, using [gvm](https://github.com/moovweb/gvm/) and `go1.9.1`, just running
go get -u github.com/cloudflare/cfssl/cmd/cfssl successfully installs cfssl successfully.

**Comment by szanwar1** [comment]
I am getting the same error even with the brew install cfssl.. can anyone tell how to resolve this

**Comment by tokiwinter** [comment]
@szanwar1 - install it yourself with `go get` as per @anthcor's comment.

**Comment by szanwar1** [comment]
@tokiwinter I have tried that, but same issue.

**Comment by tokiwinter** [comment]
@szanwar1 - I did `brew install go` which installed `go` version 1.9.1, then set up my `GOPATH`, did the `go get` as per above, and all is well.

**Comment by saran-007** [comment]
Getting the same error...

System Version:	macOS 10.13.1 (17B48)
Kernel Version:	Darwin 17.2.0

Installed `brew install go`

Installed `brew install cfssl`

'go version go1.9.2 darwin/amd64'

`go env`

OARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/sara9520/go"
GORACE=""
GOROOT="/usr/local/Cellar/go/1.9.2/libexec"
GOTOOLDIR="/usr/local/Cellar/go/1.9.2/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/xd/hd36ytsx6dx5kmnkkqyd9cf9np1mcv/T/go-build623617457=/tmp/go-build -gno-record-gcc-switches -fno-common"
CXX="clang++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"

Stack Trace when using `cfssl`

failed MSpanList_Insert 0xa0f000 0x136167bbe26cb 0x0 0x0
fatal error: MSpanList_Insert

runtime stack:
runtime.throw(0x6bbbe0, 0x10)
        /usr/local/go/src/runtime/panic.go:530 +0x90 fp=0x7ffeefbff450 sp=0x7ffeefbff438
runtime.(*mSpanList).insert(0x9436e8, 0xa0f000)
        /usr/local/go/src/runtime/mheap.go:933 +0x293 fp=0x7ffeefbff480 sp=0x7ffeefbff450
runtime.(*mheap).freeSpanLocked(0x942ee0, 0xa0f000, 0x100, 0x0)
        /usr/local/go/src/runtime/mheap.go:809 +0x4be fp=0x7ffeefbff4e8 sp=0x7ffeefbff480
runtime.(*mheap).grow(0x942ee0, 0x8, 0x0)
        /usr/local/go/src/runtime/mheap.go:675 +0x2a0 fp=0x7ffeefbff540 sp=0x7ffeefbff4e8
runtime.(*mheap).allocSpanLocked(0x942ee0, 0x1, 0x0)
        /usr/local/go/src/runtime/mheap.go:553 +0x4e3 fp=0x7ffeefbff598 sp=0x7ffeefbff540
runtime.(*mheap).alloc_m(0x942ee0, 0x1, 0x15, 0x0)
        /usr/local/go/src/runtime/mheap.go:437 +0x119 fp=0x7ffeefbff5c8 sp=0x7ffeefbff598
runtime.(*mheap).alloc.func1()
        /usr/local/go/src/runtime/mheap.go:502 +0x41 fp=0x7ffeefbff5f8 sp=0x7ffeefbff5c8
runtime.systemstack(0x7ffeefbff618)
        /usr/local/go/src/runtime/asm_amd64.s:307 +0xab fp=0x7ffeefbff600 sp=0x7ffeefbff5f8
runtime.(*mheap).alloc(0x942ee0, 0x1, 0x10000000015, 0xee2f)
        /usr/local/go/src/runtime/mheap.go:503 +0x63 fp=0x7ffeefbff648 sp=0x7ffeefbff600
runtime.(*mcentral).grow(0x944ae0, 0x0)
        /usr/local/go/src/runtime/mcentral.go:209 +0x93 fp=0x7ffeefbff6b0 sp=0x7ffeefbff648
runtime.(*mcentral).cacheSpan(0x944ae0, 0x93d4e8)
        /usr/local/go/src/runtime/mcentral.go:89 +0x47d fp=0x7ffeefbff6f0 sp=0x7ffeefbff6b0
runtime.(*mcache).refill(0xa0b000, 0x15, 0x7ffeefbff758)
        /usr/local/go/src/runtime/mcache.go:119 +0xcc fp=0x7ffeefbff728 sp=0x7ffeefbff6f0
runtime.mallocgc.func2()
        /usr/local/go/src/runtime/malloc.go:642 +0x2b fp=0x7ffeefbff748 sp=0x7ffeefbff728
runtime.systemstack(0x7ffeefbff7e8)
        /usr/local/go/src/runtime/asm_amd64.s:307 +0xab fp=0x7ffeefbff750 sp=0x7ffeefbff748
runtime.mallocgc(0x180, 0x6558a0, 0x0, 0x800000000)
        /usr/local/go/src/runtime/malloc.go:643 +0x869 fp=0x7ffeefbff828 sp=0x7ffeefbff750
runtime.newobject(0x6558a0, 0x93da10)
        /usr/local/go/src/runtime/malloc.go:781 +0x42 fp=0x7ffeefbff850 sp=0x7ffeefbff828
runtime.malg(0x8000, 0x93df00)
        /usr/local/go/src/runtime/proc.go:2634 +0x27 fp=0x7ffeefbff888 sp=0x7ffeefbff850
runtime.mpreinit(0x93e760)
        /usr/local/go/src/runtime/os1_darwin.go:140 +0x1f fp=0x7ffeefbff8a0 sp=0x7ffeefbff888
runtime.mcommoninit(0x93e760)
        /usr/local/go/src/runtime/proc.go:494 +0x105 fp=0x7ffeefbff8e8 sp=0x7ffeefbff8a0
runtime.schedinit()
        /usr/local/go/src/runtime/proc.go:434 +0x79 fp=0x7ffeefbff930 sp=0x7ffeefbff8e8
runtime.rt0_go(0x7ffeefbff968, 0x1, 0x7ffeefbff968, 0x0, 0x0, 0x1, 0x7ffeefbffad8, 0x0, 0x7ffeefbffade, 0x7ffeefbffb1a, ...)
        /usr/local/go/src/runtime/asm_amd64.s:138 +0x132 fp=0x7ffeefbff938 sp=0x7ffeefbff930

**Comment by saran-007** [comment]
@szanwar1 - I had the same issue. Added the go path to my $PATH and that did resolve. cfssl was under $HOME/go/bin for me.

I added it to my $PATH by `export PATH=$HOME/go/bin:$PATH`

**Comment by daljeetv** [comment]
@anthcor and @saran-007's combined solutions worked for me. 
`go get -u github.com/cloudflare/cfssl/cmd/cfssl`
`export PATH=$HOME/go/bin:$PATH`

**Comment by martinmosegaard** [comment]
The issue is also reported here:
https://github.com/cloudflare/cfssl/issues/813

I used `brew install cfssl` instead.

**Comment by mhumeSF** [comment]
Had this issue on high sierra, with both `cfssl` and `cfssljson`. Using `go get -u github.com/cloudflare/cfssl/cmd/cfssl` and verifying `which cfssl cfssljson` was referencing go binaries in `~/go/bin` fixed the issue for me.

**Comment by elsmorian** [comment]
If you don't have brew installed and would rather not use it just to install Go, you can use the MacOS go installer from https://golang.org/dl/ and then use the `go get` lines as above to install cfssl:

    go get -u github.com/cloudflare/cfssl/cmd/cfssl
    go get -u github.com/cloudflare/cfssl/cmd/cfssljson
    export PATH=$HOME/go/bin:$PATH
    cfssl version

Cloudflare themselves mention installing via go get also, on the project README.

**Comment by szanwar1** [comment]
thank you everyone, export PATH=$HOME/go/bin:$PATH helped

**Comment by yaron-idan** [comment]
I've proposed a fix in #308 , waiting for it to be merged.

**Comment by kelseyhightower** [comment]
I've added pointers to brew in the client tools section. Fixed on the 1.10.2 branch.
